### PR TITLE
Aliquot Sum, Narcissistic Number (Armstrong Number), Perfect Cube, Perfect Square, Perfect Number added

### PR DESCRIPTION
**Pull request by simolab99**
Aliquot Sum, Narcissistic Number (Armstrong Number), Perfect Cube, Perfect Square, Perfect Number added.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the comments
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms.Tests/Numeric/NarcissisticNumberTest.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        public static void NarcissisticNumberWork(int number)
+        {
+            // Arrange
+
+            // Act
+            var result = NarcissisticNumber.IsNarcissistic(number);
+
+            // Assert
+            Assert.IsTrue(result);
+        }
```
Comment (siriak) [inline Algorithms.Tests/Numeric/NarcissisticNumberTest.cs:22]: Please add tests to check numbers that are not narcissistic

File: Algorithms.Tests/Numeric/PerfectCubeTests.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        public static void PerfectCubeWork(int number)
+        {
+            // Arrange
+
+            // Act
+            var result = PerfectCube.IsPerfectCube(number);
+            
+            // Assert
+            Assert.IsTrue(result);
+        }
```
Comment (siriak) [inline Algorithms.Tests/Numeric/PerfectCubeTests.cs:22]: The same here, add negative cases

File: Algorithms.Tests/Numeric/PerfectNumberTest.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        public static void PerfectNumberWork(int number)
+        {
+            // Arrange
+
+            // Act
+            var result = PerfectNumber.IsPerfectNumber(number);
+
+            // Assert
+            Assert.IsTrue(result);
+        }
```
Comment (siriak) [inline Algorithms.Tests/Numeric/PerfectNumberTest.cs:22]: And here

File: Algorithms.Tests/Numeric/PerfectSquareTest.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        public static void PerfectSquareWork(int number)
+        {
+            // Arrange
+
+            // Act
+            var result = PerfectSquare.IsPerfectSquare(number);
+
+            // Assert
+            Assert.IsTrue(result);
+        }
```
Comment (siriak) [inline Algorithms.Tests/Numeric/PerfectSquareTest.cs:22]: And here

File: Algorithms/Numeric/Aliquot.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Algorithms.Numerics
+{
+    /// <summary>
+    /// In number theory, the aliquot sum s(n) of a positive integer n is the sum of all proper divisors
+    /// of n, that is, all divisors of n other than n itself.For example, the proper divisors of 15
```
Comment (siriak) [inline Algorithms/Numeric/Aliquot.cs:7]:
```suggestion
    /// of n, that is, all divisors of n other than n itself. For example, the proper divisors of 15

```

File: Algorithms/Numeric/NarcissisticNumber.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    /// <summary>
+    /// A Narcissistic number is equal to the sum of the cubes of its digits. For example, 370 is a
+    /// Narcissistic number because 3*3*3 + 7*7*7 + 0*0*0 = 370.
+    /// </summary>
+    public static class NarcissisticNumber
+    {
+        /// <summary>
+        /// Checks if a number is a Narcissistic number or not.
+        /// </summary>
+        /// <param name="number">Number too check.</param>
```
Comment (siriak) [inline Algorithms/Numeric/NarcissisticNumber.cs:14]:
```suggestion
        /// <param name="number">Number to check.</param>

```

File: Algorithms/Numeric/PerfectCube.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+{
+    /// <summary>
+    /// A perfect cube is an element of algebraic structure that is equal to the cube of another element.
+    /// </summary>
+    public static class PerfectCube
+    {
+        /// <summary>
+        /// Checks if a number is a perfect cube or not.
+        /// </summary>
+        /// <param name="number">Number too check.</param>
```
Comment (siriak) [inline Algorithms/Numeric/PerfectCube.cs:13]:
```suggestion
        /// <param name="number">Number to check.</param>

```

File: Algorithms/Numeric/PerfectCube.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        /// <returns>True if is a perfect cube; False otherwise.</returns>
+        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
+        public static bool IsPerfectCube(int number)
+        {
+            if (number < 0)
+            {
+                throw new ArgumentException(string.Format("Error number is not on interval (0.0; {0}).", int.MaxValue));
+            }
+
```
Comment (siriak) [inline Algorithms/Numeric/PerfectCube.cs:22]:
```suggestion
        public static bool IsPerfectCube(int number)
        {
```

Comment (siriak) [reply Algorithms/Numeric/PerfectCube.cs:22]: Perfect cubes can be negative AFAIK

File: Algorithms/Numeric/PerfectNumber.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        /// Checks if a number is a perfect number or not.
+        /// </summary>
+        /// <param name="number">Number too check.</param>
+        /// <returns>True if is a perfect number; False otherwise.</returns>
+        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
+        public static bool IsPerfectNumber(int number)
+        {
+            if (number < 0)
+            {
+                throw new ArgumentException(string.Format("Error number is not on interval (0.0; {0}).", int.MaxValue));
```
Comment (siriak) [inline Algorithms/Numeric/PerfectNumber.cs:22]:
```suggestion
        /// <param name="number">Number too check.</param>
        /// <returns>True if is a perfect number; False otherwise.</returns>
        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
        public static bool IsPerfectNumber(int number)
        {
            if (number < 0)
            {
                throw new ArgumentException($"{nameof(number)} cannot be negative");

```

File: Algorithms/Numeric/PerfectSquare.cs [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        /// Checks if a number is a perfect square or not.
+        /// </summary>
+        /// <param name="number">Number too check.</param>
+        /// <returns>True if is a perfect square; False otherwise.</returns>
+        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
+        public static bool IsPerfectSquare(int number)
+        {
+            if (number < 0)
+            {
+                throw new ArgumentException(string.Format("Error number is not on interval (0.0; {0}).", int.MaxValue));
```
Comment (siriak) [inline Algorithms/Numeric/PerfectSquare.cs:20]:
```suggestion
                return false;

```

**Review by siriak — COMMENTED** [review]
Hi, looks good. Could you add some more tests and address some comments I left? I cannot merge unless all the checks pass.

**Comment by simolab99** [comment]
I corrected all the above and deleted PerfectCube.cs and PerfectCubeTest.cs because I'm not sure how to get it working with the current Math.Pow() without resulting in a NaN

File: Algorithms/Numeric/Aliquot.cs [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        /// Finds the aliquot sum of an integer number.
+        /// </summary>
+        /// <param name="number">Positive number.</param>
+        /// <returns>The Aliquot Sum.</returns>
+        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
+        public static int AliquotSum(int number)
+        {
+            if (number < 0)
+            {
+                throw new ArgumentException(string.Format("Error number is not on interval (0.0; {0}).", int.MaxValue));
```
Comment (siriak) [inline Algorithms/Numeric/Aliquot.cs:23]:
```suggestion
                throw new ArgumentException(string.Format($"{nameof(number)} cannot be negative"));

```

**Review by siriak — APPROVED** [review]
Looks good, thanks!

File: Algorithms/Numeric/Aliquot.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        /// </summary>
+        /// <param name="number">Positive number.</param>
+        /// <returns>The Aliquot Sum.</returns>
+        /// <exception cref="ArgumentException">Error number is not on interval (0.0; int.MaxValue).</exception>
+        public static int AliquotSum(int number)
+        {
+            if (number < 0)
+            {
+                throw new ArgumentException($"{nameof(number)} cannot be negative");
+
```
Comment (siriak) [inline Algorithms/Numeric/Aliquot.cs:24]:
```suggestion

```

File: Algorithms/Numeric/NarcissisticNumber.cs [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+            }
+
+            temp = number;
+            while (number > 0)
+            {
+                int remainder = number % 10;
+                int power = 1;
+                for (int i = 1; i <= numberOfDigits; power *= remainder, ++i)
+                {
+                }
```
Comment (siriak) [inline Algorithms/Numeric/NarcissisticNumber.cs:34]:
```suggestion
                int power = (int)Math.Pow(remainder, numberOfDigits);

```

**Review by siriak — APPROVED** [review]
Could you add some tests to cover input of negative numbers for Aliquot sum and perfect numbers? Test coverage is too low, cannot merge until checks pass.

**Comment by simolab99** [comment]
Now all the checks pass but I don't understand why codecov/patch throws a warning on a covered line.

**Review by siriak — APPROVED** [review]
Looks good, thanks!
